### PR TITLE
[LF] Kill expirmental optimized text primitives

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
Those have never proved their usefullness.

### CODE REVIEW

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala [hunk 1]
```diff
@@ -2397,15 +2397,7 @@ private[archive] object DecodeV1 {
...
-      BuiltinFunctionInfo(TYPE_REP_TYCON_NAME, BTypeRepTyConName, minVersion = LV.v1_dev),
-      BuiltinFunctionInfo(TEXT_TO_UPPER, BTextToUpper, minVersion = unstable),
-      BuiltinFunctionInfo(TEXT_TO_LOWER, BTextToLower, minVersion = unstable),
-      BuiltinFunctionInfo(TEXT_SLICE, BTextSlice, minVersion = unstable),
-      BuiltinFunctionInfo(TEXT_SLICE_INDEX, BTextSliceIndex, minVersion = unstable),
-      BuiltinFunctionInfo(TEXT_CONTAINS_ONLY, BTextContainsOnly, minVersion = unstable),
-      BuiltinFunctionInfo(TEXT_REPLICATE, BTextReplicate, minVersion = unstable),
-      BuiltinFunctionInfo(TEXT_SPLIT_ON, BTextSplitOn, minVersion = unstable),
-      BuiltinFunctionInfo(TEXT_INTERCALATE, BTextIntercalate, minVersion = unstable),
+      BuiltinFunctionInfo(TYPE_REP_TYCON_NAME, BTypeRepTyConName, minVersion = unstable),
```
Comment (nickchapman-da) [inline daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala:2400]: An unrelated tiny cleanup slipped into this PR.

**Review by nickchapman-da — APPROVED** [review]
LGTM, thanks!

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala:2400]: Yes.
